beluga_amcl/cmake/ament.cmake

```diff
@@ -22,7 +22,8 @@ find_package(rclcpp_lifecycle REQUIRED)
 find_package(std_srvs REQUIRED)
 
 add_library(amcl_node_component SHARED)
-target_sources(amcl_node_component PRIVATE src/amcl_node.cpp)
+target_sources(amcl_node_component PRIVATE src/amcl_node.cpp
+                                           src/ros2_common.cpp)
```

Comment:
@serraramiro1 why do we compile `ros2_common.cpp` twice? This calls for a shared library :)

Reply:
Done! Initially faced some issues with ament macros but eventually got it right